client/layout/command-palette.tsx

```diff
@@ -0,0 +1,58 @@
+import CommandPalette from '@automattic/command-palette';
+import { useSiteExcerptsSorted } from 'calypso/data/sites/use-site-excerpts-sorted';
+import { navigate } from 'calypso/lib/navigate';
+import { useCommandsCalypso } from 'calypso/sites-dashboard/components/wpcom-smp-commands';
+import { useSelector } from 'calypso/state';
+import { closeCommandPalette } from 'calypso/state/command-palette/actions';
+import { isCommandPaletteOpen as getIsCommandPaletteOpen } from 'calypso/state/command-palette/selectors';
+import { getCurrentRoutePattern } from 'calypso/state/selectors/get-current-route-pattern';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import type { CommandPaletteProps } from '@automattic/command-palette';
+
+interface CurrentUserCapabilitiesState {
+	currentUser: {
+		capabilities: {
+			[ key: number ]: { [ key: string ]: boolean };
+		};
+	};
+}
+
+// This selector is a bit too broad, but is needed to match the interface on CommandPalette.
+// We're therefore keeping it private here, instead of making it available across Calypso.
+const getCurrentUserCapabilities = ( state: CurrentUserCapabilitiesState ) =>
+	state.currentUser.capabilities;
+
+const CalypsoCommandPalette = (
+	props: Omit<
+		CommandPaletteProps,
+		| 'currentRoute'
+		| 'currentSiteId'
+		| 'isOpenGlobal'
+		| 'onClose'
+		| 'navigate'
+		| 'useCommands'
+		| 'useSites'
+		| 'userCapabilities'
+	>
+) => {
+	const isCommandPaletteOpen = useSelector( getIsCommandPaletteOpen );
+	const currentRoutePattern = useSelector( getCurrentRoutePattern ) ?? '';
+	const currentSiteId = useSelector( getSelectedSiteId );
+	const userCapabilities = useSelector( getCurrentUserCapabilities );
+
+	return (
+		<CommandPalette
+			currentRoute={ currentRoutePattern }
+			currentSiteId={ currentSiteId }
+			isOpenGlobal={ isCommandPaletteOpen }
+			onClose={ closeCommandPalette }
+			navigate={ navigate }
+			useCommands={ useCommandsCalypso }
+			useSites={ useSiteExcerptsSorted }
+			userCapabilities={ userCapabilities }
+			{ ...props }
```

Comment:
Nitpick (non-blocking): We can further simplify this by removing the external props here since we no longer need to pass any prop from the layout component.

```diff
diff --git a/client/layout/command-palette.tsx b/client/layout/command-palette.tsx
index 8ce8bc5f51..9cbfa5284d 100644
--- a/client/layout/command-palette.tsx
+++ b/client/layout/command-palette.tsx
@@ -7,7 +7,6 @@ import { closeCommandPalette } from 'calypso/state/command-palette/actions';
 import { isCommandPaletteOpen as getIsCommandPaletteOpen } from 'calypso/state/command-palette/selectors';
 import { getCurrentRoutePattern } from 'calypso/state/selectors/get-current-route-pattern';
 import { getSelectedSiteId } from 'calypso/state/ui/selectors';
-import type { CommandPaletteProps } from '@automattic/command-palette';
 
 interface CurrentUserCapabilitiesState {
 	currentUser: {
@@ -22,19 +21,7 @@ interface CurrentUserCapabilitiesState {
 const getCurrentUserCapabilities = ( state: CurrentUserCapabilitiesState ) =>
 	state.currentUser.capabilities;
 
-const CalypsoCommandPalette = (
-	props: Omit<
-		CommandPaletteProps,
-		| 'currentRoute'
-		| 'currentSiteId'
-		| 'isOpenGlobal'
-		| 'onClose'
-		| 'navigate'
-		| 'useCommands'
-		| 'useSites'
-		| 'userCapabilities'
-	>
-) => {
+const CalypsoCommandPalette = () => {
 	const isCommandPaletteOpen = useSelector( getIsCommandPaletteOpen );
 	const currentRoutePattern = useSelector( getCurrentRoutePattern ) ?? '';
 	const currentSiteId = useSelector( getSelectedSiteId );
@@ -50,7 +37,6 @@ const CalypsoCommandPalette = (
 			useCommands={ useCommandsCalypso }
 			useSites={ useSiteExcerptsSorted }
 			userCapabilities={ userCapabilities }
-			{ ...props }
 		/>
 	);
 };

```

Reply:
Thanks, @mmtr! I went ahead and did that.

The reason I hadn't is because there are other props in `CommandPalette` we might want to use, such as `selectedCommand`. But I guess we can always add those in if we need them, later on 👍